src/compiler/checker.ts

```diff
@@ -1558,16 +1559,16 @@ namespace ts {
                     }
                 }
 
-                function writeTypeList(types: Type[], union: boolean) {
+                function writeTypeList(types: Type[], delimiter: SyntaxKind) {
                     for (let i = 0; i < types.length; i++) {
                         if (i > 0) {
-                            if (union) {
+                            if (delimiter !== SyntaxKind.CommaToken) {
                                 writeSpace(writer);
                             }
-                            writePunctuation(writer, union ? SyntaxKind.BarToken : SyntaxKind.CommaToken);
+                            writePunctuation(writer, delimiter);
                             writeSpace(writer);
                         }
-                        writeType(types[i], union ? TypeFormatFlags.InElementType : TypeFormatFlags.None);
+                        writeType(types[i], delimiter === SyntaxKind.CommaToken ? TypeFormatFlags.None : TypeFormatFlags.InElementType);

```

Comment:
Oh right. In that case, can you change the name InElementType to something else? That's what confused me. What this flag is actually trying to say is that the type list is itself a type.